### PR TITLE
Fix #2529: Fix username's cutoff in Various screens

### LINKED ISSUE #2529: Check and fix Profile Name across all screens
Profile Name is shown in below mentioned screens and we should make sure that a name with maximum characters is clearly visible.
- Profile Chooser Screen
- Navigation Drawer Header
- Home Screen Salutaion
- Profile Progress
- Profile List (in Admin Controls)
- Profile View (in Admin Controls)

These needs to be tested and fixed on mobile-landscape, mobile-portrait, tablet-landscape and  tablet-portrait.

### PR DESCRIPTION
**Pull request by khushal1707**
## Explanation
Fixes #2529: Fix username's cutoff in:

- ~~Profile Chooser Screen on mobile-landscape, mobile-portrait, tablet-landscape and tablet-portrait~~
- Profile Progress on mobile-landscape, mobile-portrait
- Profile View (in Admin Controls) on mobile-portrait
- Profile List (in Admin Controls) on mobile-portrait, mobile-landscape

| Before           |  After |
:-------------------------:|:-------------------------:
Screenshot_20210122-093455 | Screenshot_20210125-220641
Screenshot_20210122-093648 | Screenshot_20210125-220710
Screenshot_20210127-100206 | Screenshot_20210127-100331

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by FareesHussain** [comment]
@khushal1707 please update your PR title and description as mentioned in the checklist
PR title ->
`Fix #2529: Fix username's cutoff in Various screens`

PR description ->
change `Fix#2529` to `Fixes #2529`

File: app/src/main/res/layout-land/profile_chooser_profile_view.xml [hunk 1]
```diff
@@ -52,10 +52,9 @@
         <TextView
           android:id="@+id/profile_name_text"
           style="@style/Caption"
-          android:ellipsize="end"
+          android:layout_width="wrap_content"
+          android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-land/profile_chooser_profile_view.xml]:
```
android:layout_width="wrap_content"
android:layout_height="wrap_content"
```

Both the above attributes are already part of `Caption` style so remove them.

File: app/src/main/res/layout-sw600dp-port/profile_chooser_profile_view.xml [hunk 2]
```diff
@@ -60,10 +60,9 @@
         <TextView
           android:id="@+id/profile_name_text"
           style="@style/Heading2"
-          android:ellipsize="end"
+          android:layout_width="wrap_content"
+          android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-port/profile_chooser_profile_view.xml]:
```
android:layout_width="wrap_content"
android:layout_height="wrap_content"
```

Both the above attributes are already part of `Caption` style so remove them.

File: app/src/main/res/layout-sw600dp-land/profile_chooser_profile_view.xml [hunk 3]
```diff
@@ -59,10 +59,9 @@
         <TextView
           android:id="@+id/profile_name_text"
           style="@style/Heading2"
-          android:ellipsize="end"
+          android:layout_width="wrap_content"
+          android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/profile_chooser_profile_view.xml]:
```
android:layout_width="wrap_content"
android:layout_height="wrap_content"
```

Both the above attributes are already part of `Caption` style so remove them.

File: app/src/main/res/layout/profile_chooser_profile_view.xml [hunk 4]
```diff
@@ -54,11 +54,10 @@
 
         <TextView
           android:id="@+id/profile_name_text"
+          android:layout_width="wrap_content"
+          android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_chooser_profile_view.xml]:
```
android:layout_width="wrap_content"
android:layout_height="wrap_content"
```

Both the above attributes are already part of `Caption` style so remove them.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@khushal1707 Please add screenshots to show that the name is fully displayed.
Note: In issues related to UI always add screenshot for reference. Thanks.

**Comment by khushal1707** [comment]
@rt4914 Please review this PR and give suggestion for changes if any.

**Comment by anandwana001** [comment]
Just adding a point here, @khushal1707 could you try adding 4-5 profiles and then check how does the long name profile look on the Profile Choser screen. 
As the height of the item increasing, just need to confirm does it affect the UI of the list on the profile chooser screen.

**Comment by khushal1707** [comment]
> Just adding a point here, @khushal1707 could you try adding 4-5 profiles and then check how does the long name profile look on the Profile Choser screen.
> As the height of the item increasing, just need to confirm does it affect the UI of the list on the profile chooser screen.

Are these looking good?

**Comment by anandwana001** [comment]
putting this screenshot just for reference in how the UI looks if we have the name at full capacity.

**Comment by khushal1707** [comment]
This is how name with max character looking along with smaller name:



Should I revert this change?

**Comment by rt4914** [comment]
@mschanteltc Please have a look at just above comment. Based on that should be show the full name in `Profile Chooser` or should we limit that to `2` lines?

**Comment by mschanteltc** [comment]
> @mschanteltc Please have a look at just above comment. Based on that should be show the full name in `Profile Chooser` or should we limit that to `2` lines?

Let's limit it to 2 lines and ellipsize the names if they exceed this limit.

**Comment by khushal1707** [comment]
@rt4914 Please review this PR again, thanks.

File: app/src/main/res/layout-land/profile_progress_header.xml [hunk 5]
```diff
@@ -46,12 +46,16 @@
...
+      android:paddingEnd="185dp"
       android:layout_marginTop="12dp"
       android:fontFamily="sans-serif-medium"
       android:text="@{viewModel.profile.name}"
       android:textColor="@color/oppiaPrimaryText"
       android:textSize="28sp"
-      app:layout_constraintStart_toStartOf="@+id/ongoing_topics_container"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintHorizontal_bias="0.0"
+      app:layout_constraintStart_toEndOf="@+id/profile_edit_image"
```
Comment (rt4914) [inline app/src/main/res/layout-land/profile_progress_header.xml]: Please don't introduce random margin and padding. In oppia every size of every item is dependent upon mocks and we try to make it match with that as much as possible.

To fix UI for this screen you can use below code:
```
 <TextView
      android:id="@+id/profile_name_text_view"
      android:layout_width="0dp"
      android:layout_height="wrap_content"
      android:layout_marginTop="12dp"
      android:fontFamily="sans-serif-medium"
      android:text="@{viewModel.profile.name}"
      android:textColor="@color/oppiaPrimaryText"
      android:textSize="28sp"
      app:layout_constraintEnd_toEndOf="@id/completed_stories_container"
      app:layout_constraintStart_toStartOf="@+id/ongoing_topics_container"
      app:layout_constraintTop_toTopOf="parent" />
```

File: app/src/main/res/layout/profile_edit_activity.xml [hunk 6]
```diff
@@ -210,6 +210,7 @@
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:fontFamily="sans-serif-medium"
+            android:layout_marginEnd="150dp"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_edit_activity.xml]: This is incorrect. Please do not introduce random margin/padding in UI.

To fix for this use following thing:
In line `197` update width to `android:layout_width="0dp"`
and in line `210` update width to `android:layout_width="match_parent"`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Along with the below mentioned changes there is UI issue in `portrait - profile progress` too.



In this above image the profile name should be bounded between the `left-most-border` of `Topics in Progress` and `right-most-border` of `Completed Stories` boxes.

File: app/src/main/res/layout/profile_edit_activity.xml (hunk 6 above)
Comment (khushal1707) [reply app/src/main/res/layout/profile_edit_activity.xml]: Thanks for the help!

File: app/src/main/res/layout-land/profile_progress_header.xml (hunk 5 above)
Comment (khushal1707) [reply app/src/main/res/layout-land/profile_progress_header.xml]: Thanks for the help!

**Comment by khushal1707** [comment]
This is how its looking after change:

**Comment by khushal1707** [comment]
@rt4914 PTAL

File: app/src/main/res/layout-land/profile_progress_header.xml (hunk 5 above)
Comment (rt4914) [reply app/src/main/res/layout-land/profile_progress_header.xml]: @khushal1707 Please do not resolve comments. The person who has started the comment thread should resolve it.

File: app/src/main/res/layout/profile_edit_activity.xml (hunk 6 above)
Comment (rt4914) [reply app/src/main/res/layout/profile_edit_activity.xml]: @khushal1707 Please do not resolve comments. The person who has started the comment thread should resolve it.

**Review by rt4914 — APPROVED** [review]
Nice work. Thanks

**Pull request merged**